lf/demo/demo_robot.lf

```diff
@@ -44,7 +73,13 @@ reactor Sensors {
 /**
  * The controller for the state machine
  */
+// This is what we call app containers. App containers can't have any direct access to hardware but can work in any machine or board. They use drivers to do things.
 reactor Controller {
+    // This is interesting because here you are describing instance of external shared value needed by the Controller App. Or the input need of the App.
+    // As I mentioned previously Drivers can't send informations by themself, app have to ask them to do it.
+    // Also Apps have to find the good container on the system to fill its need (by Alias, type, ID, or physical position for example).
```

Comment:
So is an app a higher-level object than a container? Does an app have multiple containers?

Reply:
Maybe an `app` corresponds to `main reactor` in LF.

Reply:
An app is a normal container technically completely similar to a driver.
The difference is just what you are doing in this container.
A Driver is a dumb container unable to control any other, it just share values of a physical feature.
An App is an intelligent container. Apps are aware of other containers and can use them. Apps can't have any dependance to hardware (except some generic information such as time, or msg or stuffs like that).

You can move an App container to any board of a distributed system without any difference regarding the behavior.
You can't move a Driver container because it depend on a specifc hardware.